e2etests/test/server_valid.js

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var serviceSdk = require('azure-iothub');
+var serviceSas = require('azure-iothub').SharedAccessSignature;
+var provisioningServiceClient = require('azure-iot-provisioning-service').ProvisioningServiceClient;
+var Message = require('azure-iot-common').Message;
+var anHourFromNow = require('azure-iot-common').anHourFromNow;
+var Registry = require('azure-iothub').Registry;
+var NoRetry = require('azure-iot-common').NoRetry;
+var deviceSdk = require('azure-iot-device');
+
+
+var uuid = require('uuid');
+
+var hubConnectionString = process.env.IOTHUB_CONN_STRING_INVALIDCERT;
+var deviceConnectionString = process.env.IOTHUB_DEVICE_CONN_STRING_INVALIDCERT;
+var dpsDeviceClientEndpoint = process.env.DPS_GLOBALDEVICEENDPOINT_INVALIDCERT;
+
+
+var correctDisconnectMessage = function(err, done) {
+  if (err) {
+    if (err.amqpError && (err.amqpError.name === 'NotConnectedError')) {
+      done();
+    } else if (err.name && (err.name  === 'NotConnectedError')) {
+      done();
+    } else if (err.code && (err.code === 'UNABLE_TO_VERIFY_LEAF_SIGNATURE')) {
+      done();
+    } else {
+      done(new Error('client did NOT detect bad cert.'));
+    }
+  } else {
+    done(new Error('client did NOT detect bad cert.'));
+  }
+};
+
+
+describe('Service Client', function () {
+  this.timeout(60000);
+  [
+    require('azure-iothub').Amqp,
```

Comment:
why deviate from the convention of having all require statements at the top of the file? makes it harder to see at first glance what the tests depend on.

Reply:
Because I find it irritating to simply create a variable that will ONLY be used to initialize this array.

---
In reply to: [220400541](https://github.com/Azure/azure-iot-sdk-node/pull/378#discussion_r220400541) [](ancestors = 220400541)